src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -393,16 +394,23 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
     if (this.isCommunityContribution || this.isUnlimitedLiability) {
       return true
     }
-    // Named ULC to BC Limited require a name request.
+    // Named ULC to BC Limited (or Benefit Company) requires a name request.
     if (this.getOriginalLegalType === CorpTypeCd.BC_ULC_COMPANY &&
-      (this.isBcLimited || this.isBenefitCompany)) {
```

Comment:
I think checking `isBenefitCompany` here was incorrect. That's a store getter that looks at the current entity type.

I think, instead, this should have checked the `isBenefit`. That's a computed that looks at the selected entity type (same as isBcLimited, isCommunityContribution, etc).